File: app/src/main/java/com/merino/ddfilms/adapters/CrewAdapter.java
package com.merino.ddfilms.adapters;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.merino.ddfilms.R;
import com.merino.ddfilms.model.Credits;

import java.util.List;

import lombok.Setter;

@Setter
public class CrewAdapter extends RecyclerView.Adapter<CrewAdapter.CrewViewHolder> {

    private List<Credits.Crew> crewList;

    public CrewAdapter(List<Credits.Crew> crewList) {
        this.crewList = crewList;
    }

    @NonNull
    @Override
    public CrewViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_person, parent, false);
        return new CrewViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull CrewViewHolder holder, int position) {
        Credits.Crew crew = crewList.get(position);

        // Configura el nombre y el personaje
        holder.personName.setText(crew.getName());
        holder.personRole.setText(crew.getJob());

        // Comprobar si el profilePath no es null antes de cargar la imagen
        if (crew.getProfilePath() != null) {
            Glide.with(holder.itemView.getContext())
                    .load("https://image.tmdb.org/t/p/w500/" + crew.getProfilePath())
                    .placeholder(R.drawable.ic_profile) // Imagen por defecto mientras carga
                    .error(R.drawable.ic_close) // Imagen si ocurre un error al cargar
                    .into(holder.personImage);
        } else {
            // Si no hay profilePath, usa una imagen por defecto
            holder.personImage.setImageResource(R.drawable.ic_profile);
        }
    }

    @Override
    public int getItemCount() {
        return crewList != null ? crewList.size() : 0;
    }

    static class CrewViewHolder extends RecyclerView.ViewHolder {
        ImageView personImage;
        TextView personName;
        TextView personRole;

        public CrewViewHolder(@NonNull View itemView) {
            super(itemView);
            personImage = itemView.findViewById(R.id.person_image);
            personName = itemView.findViewById(R.id.person_name);
            personRole = itemView.findViewById(R.id.person_role);
        }
    }
}

File: app/src/main/java/com/merino/ddfilms/adapters/CrewAdapter.kt
package com.merino.ddfilms.adapters

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.merino.ddfilms.R
import com.merino.ddfilms.model.Credits

class CrewAdapter(
    var crewList: List<Credits.Crew>?
) : RecyclerView.Adapter<CrewAdapter.CrewViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CrewViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_person, parent, false)
        return CrewViewHolder(view)
    }

    override fun onBindViewHolder(holder: CrewViewHolder, position: Int) {
        val crewListSafe = crewList ?: return
        val crew = crewListSafe[position]

        holder.personName.text = crew.name
        holder.personRole.text = crew.job

        if (crew.profilePath != null) {
            Glide.with(holder.itemView.context)
                .load("https://image.tmdb.org/t/p/w500/" + crew.profilePath)
                .placeholder(R.drawable.ic_profile)
                .error(R.drawable.ic_close)
                .into(holder.personImage)
        } else {
            holder.personImage.setImageResource(R.drawable.ic_profile)
        }
    }

    override fun getItemCount(): Int = crewList?.size ?: 0

    class CrewViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val personImage: ImageView = itemView.findViewById(R.id.person_image)
        val personName: TextView = itemView.findViewById(R.id.person_name)
        val personRole: TextView = itemView.findViewById(R.id.person_role)
    }
}
